lib/utils.js

```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
   return res;
 }
 
+const DecoderMap = new Map();
+
 function decodeText(text, textEncoding, destEncoding) {
-  var ret;
-  if (text && jsencoding.encodingExists(destEncoding)) {
-    try {
-      ret = jsencoding.TextDecoder(destEncoding)
-                      .decode(Buffer.from(text, textEncoding));
-    } catch(e) {}
+  let ret;
+  /**
+   * https://datatracker.ietf.org/doc/html/rfc7578#section-4.7
+   */
+  switch (destEncoding) {
+    case '7bit':
+      destEncoding = 'ascii';
+      break;
+    case '8bit':
+      destEncoding = 'binary';
+      break;
+  }
+  if (text) {
```

Comment:
do we need destEncoding ourmtside of this block?

Reply:
no, we get from outside if it is 7bit or 8bit. But I think I made a mistake because this is actually the textEncoding and not the destEncoding

Reply:
considering that all tests still passed, it's a very alarming sign. we do need to improve our code coverage for this part of functionality ahead of this change.

Reply:
I think it would be a huge task to do that, with little to no benefits. Or are there any nodejs unit tests for the builtin textdecoder, which we could reuse?